frontend/src/Pages/ResourcesManagement.tsx

```diff
@@ -147,21 +214,55 @@ export default function ResourcesManagement() {
     };
 
     const updateFinalState = async (e: React.MouseEvent) => {
+        setToast({ state: ToastState.null, message: '' });
         e.preventDefault();
         const newCollectionList = collectionList.map((c, i) => {
             c.rank = i + 1;
             c.id = i;
             return c;
         });
-        const response = await API.put('left-menu', newCollectionList);
-        if (response.success) {
-            await mutate();
-            toaster('Collections Saved!', ToastState.success);
-        } else {
-            toaster(
-                'All collections must have associated links',
-                ToastState.error
+        try {
+            const response = await axios.put(
+                '/api/left-menu',
+                newCollectionList
             );
+            if (response.status !== 201) {
+                setToast({
+                    state: ToastState.error,
+                    message: 'Error Saving Collections'
+                });
+            } else {
+                await mutate();
+                setToast({
+                    state: ToastState.success,
+                    message: 'Collections Saved!'
+                });
+                setInitialCollectionList(newCollectionList);
+
+                // Update the originalLinkValues with the new saved state
+                if (selectedCollectionIndex !== undefined) {
+                    const newOriginalValues: Record<number, ResourceLink> = {};
+                    newCollectionList[selectedCollectionIndex].links.forEach(
+                        (link, index) => {
+                            newOriginalValues[index] = { ...link };
+                        }
+                    );
+                    setOriginalLinkValues(newOriginalValues);
+                }
+
+                setCollectionList(
+                    newCollectionList.map((c) => ({
+                        ...c,
+                        isModified: false
+                    }))
+                );
+            }
+            //eslint-disable-next-line
```

Comment:
If this needs to stay (it may not with the previous comment) can we remove the `eslint-disable-next-line` or explain why it is necessary?